src/ibtracert.c

```diff
@@ -811,9 +820,29 @@ int main(int argc, char **argv)
 	argc -= optind;
 	argv += optind;
 
-	if (argc < 2)
+	if (argc < 2 && lid_file[0] == '\0') 
 		ibdiag_show_usage();
 
+	if (argc == 2) {
+		srcid = strdup(argv[0]);
+		dstid = strdup(argv[1]);
```

Comment:
Check for strdup errors and IBEXIT if so

Reply:
okey dokey